src/Microsoft.Health.Dicom.Core/Features/Validation/DicomElementValidation.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Extensions;
+
+namespace Microsoft.Health.Dicom.Core.Features.Validation
+{
+    internal class DicomElementValidation : IDicomElementValidation
+    {
+        public virtual void Validate(DicomElement dicomElement)
+        {
+            EnsureArg.IsNotNull(dicomElement, nameof(dicomElement));
+            DicomVR vr = dicomElement.ValueRepresentation;
+            if (ValidationLimits.SupportedVRs.Contains(vr))
+            {
+                // only works for single value dicom element ( Since we accept empty/null value, Count = 0 is accepted).
+                if (dicomElement.Count > 1)
+                {
+                    throw new DicomElementValidationException(
+                        ValidationErrorCode.ElementHasMultipleValues,
+                        dicomElement.Tag.GetFriendlyName(),
+                        vr,
+                        DicomCoreResource.DicomElementHasMultipleValues);
+                }
+            }
+            else
+            {
+                Debug.Fail($"Validating VR {vr.Code} is not supported.");
+            }
+        }
+
+        protected static bool IsControlExceptESC(char c)
+        => char.IsControl(c) && (c != '\u001b');
```

Comment:
Furthermore, instead of defining this as a predicate, it seems like it is always used to check for characters in a `string`. Should this not simply be defined like:

```csharp
protected static bool ContainsControlExceptESC(string s)
```

Reply:
will be resolved.